language-design.md

```diff
@@ -27,7 +56,19 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
+In the sound lab there was the issue that each sound could only have a single transformation applied to it (and an unavoidable side effect of playing the sound and writing to a file).  This made it very hard to extend the library - multiple transformations couldn't be done by cleanly calling multiple methods sequentially, but instead by writing a new function that explicitly performed all of those transformations (or even worse, daisy-chaining using the fact that the output file name was "out.wav").  This violated [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as well as ease of expansion and use.  
+
+If it had been designed in the style that _Growing a Language_ recommended, however, it would be easy to make the change mentioned above (chaining sound transformations) as well as extending the module to perform new, unplanned-for things such as playing or transforming an entire musical score.  The way the language (module) is planned from the get-go has a huge impact on the way users can and will use your module, or if they will at all.
 
+> Parts of the language must be designed to help the task of growth. A good set of types,
+ways for a user to define new types, to add new words and new rules to the language, to
+define and use all sorts of patterns—all these are needed. The designer should not, for
+example, define twenty kinds of number types in the language. But there will be users
+who, all told, beg for twenty kinds of numbers. The language should have a way for the
+user to define number types that work well with each other, and with plus signs and other
+such signs, and with the many ways of pushing bits in and out of the computer.  [Steele, 1998]
+
+It wouldn't have made sense for the designer of the sound lab to make it able to work with whole scores - but it should have been designed so that someone else could make it do that if they wanted to.  In the same way that it doesn't make sense for the language designer to define 20 numeric types, the designer shouldn't have 1000 sound transforms - but it should have been doable for those users who do need that functionality.
```

Comment:
I think the important take away is to build enough to make an obvious pattern, a pattern that users can add onto using their own transformations.